File: Nadeem Ahmed/Angular Assignments/Task Management/backend/src/seeders/data.seeder.js
const bcrypt = require("bcrypt");

const {
    Role,
    User,
    Task
} = require("../models");

const seedDatabase = async () => {

    try {

        await Role.findOrCreate({
            where: { id: 1 },
            defaults: {
                name: "Admin"
            }
        });

        await Role.findOrCreate({
            where: { id: 2 },
            defaults: {
                name: "Manager"
            }
        });

        await Role.findOrCreate({
            where: { id: 3 },
            defaults: {
                name: "Employee"
            }
        });

        console.log("Roles Seeded");

        const adminPassword =
            await bcrypt.hash(
                "Admin@123",
                10
            );

        const managerPassword =
            await bcrypt.hash(
                "Manager@123",
                10
            );

        const employeePassword =
            await bcrypt.hash(
                "Employee@123",
                10
            );

        await User.findOrCreate({
            where: {
                email: "admin@example.com"
            },
            defaults: {
                username: "admin",
                password: adminPassword,
                roleId: 1
            }
        });

        await User.findOrCreate({
            where: {
                email: "manager@example.com"
            },
            defaults: {
                username: "manager",
                password: managerPassword,
                roleId: 2
            }
        });

        await User.findOrCreate({
            where: {
                email: "employee1@example.com"
            },
            defaults: {
                username: "employee1",
                password: employeePassword,
                roleId: 3
            }
        });

        await User.findOrCreate({
            where: {
                email: "employee2@example.com"
            },
            defaults: {
                username: "employee2",
                password: employeePassword,
                roleId: 3
            }
        });

        console.log("Users Seeded");

        

        const taskCount =
            await Task.count();

        if (taskCount === 0) {

            await Task.bulkCreate([
                {
                    title: "Design Login Page",
                    description:
                        "Create login page UI",
                    status: "Pending",
                    dueDate: new Date(),
                    assignedToUserId: 3,
                    assignedByUserId: 2
                },
                {
                    title: "Implement JWT",
                    description:
                        "Add JWT Authentication",
                    status: "In Progress",
                    dueDate: new Date(),
                    assignedToUserId: 4,
                    assignedByUserId: 1
                }
            ]);

            console.log("Tasks Seeded");
        }

        console.log(
            "Database Seeding Completed"
        );

    } catch (error) {

        console.error(
            "Seeder Error:",
            error
        );

    }
};

module.exports = seedDatabase;

File: Parul kumawat/.Net assignment/EmployeeTaskManagementAPI/EmployeeTaskManagementAPI/Data/DataSeeder.cs
using EmployeeTaskManagementAPI.Enum;
using EmployeeTaskManagementAPI.Models;
using Microsoft.AspNetCore.Identity;
using Microsoft.EntityFrameworkCore;

namespace EmployeeTaskManagementAPI.Data
{
    public class DataSeeder
    {
        public static async Task SeedRoles(IServiceProvider serviceProvider)
        {
            var roleManager = serviceProvider.GetRequiredService<RoleManager<IdentityRole>>();
            var userManager = serviceProvider.GetRequiredService<UserManager<ApplicationUser>>();
            var context = serviceProvider.GetRequiredService<AppDbContext>();


            string[] roles = { "Admin", "Employee", "Manager" };

            foreach (var role in roles)
            {
                if (!await roleManager.RoleExistsAsync(role))
                {
                    await roleManager.CreateAsync(new IdentityRole(role));
                }
            }

            var admin = await userManager.FindByEmailAsync("admin@gmail.com");

            if (admin == null)
            {
                admin = new ApplicationUser
                {
                    FirstName = "Admin",
                    UserName = "admin",
                    Email = "admin@gmail.com"
                };


                var result = await userManager.CreateAsync(admin, "Admin@123");
                if (result.Succeeded)
                {
                    await userManager.AddToRoleAsync(admin, "Admin");
                }
            }
            var manager = await userManager.FindByEmailAsync("manager@gmail.com");

            if (manager == null)
            {
                manager = new ApplicationUser
                {
                    FirstName = "Manager",
                    UserName = "manager",
                    Email = "manager@gmail.com"
                };

                var result1 = await userManager.CreateAsync(manager, "Manager@123");
                if (result1.Succeeded)
                {
                    await userManager.AddToRoleAsync(manager, "Manager");
                }
            }

            var emp1 = await userManager.FindByEmailAsync("emp1@gmail.com");
            if (emp1 == null)
            {
                emp1 = new ApplicationUser
                {
                    FirstName = "Employee1",
                    UserName = "emp1",
                    Email = "emp1@gmail.com"
                };

                var res = await userManager.CreateAsync(emp1, "Emp@123");
                if (res.Succeeded)
                {
                    await userManager.AddToRoleAsync(emp1, "Employee");
                }
            }


            var emp2 = await userManager.FindByEmailAsync("emp2@gmail.com");
            if (emp2 == null)
            {
                emp2 = new ApplicationUser
                {
                    FirstName = "Employee2",
                    UserName = "emp2",
                    Email = "emp2@gmail.com"
                };

                var res = await userManager.CreateAsync(emp2, "Emp@123");
                if (res.Succeeded)
                {
                    await userManager.AddToRoleAsync(emp2, "Employee");
                }
            }
            await context.SaveChangesAsync();



            if (!await context.Tasks.AnyAsync())
            {
                var tasks = new List<Tasks>
                {
                    new Tasks
                    {
                        Title = "Setup Project Structure",
                        Description = "Initialize API structure and layers",
                        Status = TasksStatus.Pending.ToString(),
                        AssignedToUserId = emp1.Id
                    },
                    new Tasks
                    {
                        Title = "Design Database Schema",
                        Description = "Create tables for users and tasks",
                        Status = TasksStatus.InProgress.ToString(),
                        AssignedToUserId = emp2.Id
                    },
                    new Tasks
                    {
                        Title = "Implement Authentication",
                        Description = "JWT authentication setup",
                        Status = TasksStatus.Pending.ToString(),
                        AssignedToUserId = emp1.Id
                    },
                    new Tasks
                    {
                        Title = "Create Task API",
                        Description = "CRUD operations for tasks",
                        Status = TasksStatus.Pending.ToString(),
                        AssignedToUserId = emp2.Id
                    },
                    new Tasks
                    {
                        Title = "Write Unit Tests",
                        Description = "Test service layer logic",
                        Status = TasksStatus.Pending.ToString(),
                        AssignedToUserId = emp1.Id
                    }
                };

                await context.Tasks.AddRangeAsync(tasks);
                await context.SaveChangesAsync();
            }
        }

    }
    }


